src/main/java/io/dapr/components/StateStore.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dapr.components;
+
+import io.dapr.components.aspects.AdvertisesFeatures;
+import io.dapr.components.aspects.InitializableWithProperties;
+import lombok.Builder;
+import lombok.NonNull;
+
+import java.util.Optional;
+
+public interface StateStore extends InitializableWithProperties, AdvertisesFeatures {
```

Comment:
I noticed that we're implementing the `bulk*` operations on the ServiceWrapper class. However  I think we should, instead, delegate bulk* operations for the component itself, some components can handle bulk operations on the underlying resource (sql,redis/etc) which could be more performant (and less error prone) than calling get one by one.

If the component doesn't support bulk operations we should have a way to fallback the implementation for the default behavior (invoke get one by one), for that I'd suggest to use default methods implemented here as the default behavior is just calling get in parallel.

```java
public class MyStateStore implements StateStore extends DefaultBulkOperations {

   default List<Optional<StateStore.Value>> bulkGet(List<String> keys) {
        // default method implementation
    }
//...
}
```

Reply:
Disclaimer: You can merge this PR without addressing this issue, it can be done with subsequent PRs if you prefer

Reply:
This was done in #17